### PR TITLE
Add E2E workflow github action

### LINKED ISSUE #340: Validate E2E workflow in CI
## Is your feature request related to a problem? Please describe.

Before adding E2E tests with more environments and authenticators, it would help to
set up either Jenkins or a GitHub action to validate the E2E workflows. Whichever CI
is used could trigger on each PR and daily to ensure additions work in an ambiguous
environment.

## Describe the solution you would like

CI flow to validate the current E2E workflow path, using either
- [ ] GitHub action
- [ ] Jenkinsfile

The CI flow should also be easily extensible to add other elements of the test matrix described in #310.

### PR DESCRIPTION
**Pull request by NaomiHaser**
### What does this PR do?
A GitHub action(CI flow) in order to validate the E2E workflows.

### What ticket does this PR close?
Resolves #340

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

#### Manual tests
**If you are preparing for a release**, have you run the following manual tests to verify existing functionality continues to function as expected?
- [ ] Manually run [Kubernetes-Conjur demo](https://github.com/conjurdemos/kubernetes-conjur-demo) with a local authn-k8s client image build

### CODE REVIEW

File: .github/workflows/e2e-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Install Helm
+        uses: azure/setup-helm@v1
+        with:
+          version: v3.3.0
+
+      - name: Run Start
+        run: bin/test-workflow/start
+
+
+
```
Comment (orenbm) [inline .github/workflows/e2e-workflow.yaml:38]: missing newline at EOF

File: .github/workflows/e2e-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  pull_request:
+    types:
+    - opened
+    - reopened
+    - synchronize
+
+jobs:
+  run-E2E-workflow:
+    name: Run E2E Workflow on KinD cluster
+    runs-on: ubuntu-latest ##macos-latest
```
Comment (orenbm) [inline .github/workflows/e2e-workflow.yaml:19]: is the `##macos-latest` comment a leftover?

File: .github/workflows/e2e-workflow.yaml (hunk 1 above)
Comment (orenbm) [reply .github/workflows/e2e-workflow.yaml:38]: also - can you remove redundant empty lines at EOF?

File: .github/workflows/e2e-workflow.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+name: E2E Workflow
+
+on:
+  # Run this on pushes to main
+  push:
+    branches:
+    - main
+
+  # Or when PR operations are done
+  pull_request:
```
Comment (orenbm) [inline .github/workflows/e2e-workflow.yaml:10]: do we want to add `branches: [ master ]` so it runs only on PRs to master?

File: .github/workflows/e2e-workflow.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+name: E2E Workflow
+
+on:
+  # Run this on pushes to main
+  push:
+    branches:
+    - main
```
Comment (orenbm) [inline .github/workflows/e2e-workflow.yaml:7]: so will this happen when the PR is merged to main?

**Comment by orenbm** [comment]
@NaomiHaser i noticed that this file has a very clear name that says what it does (which is great!). Do you think we should rename `main.yml` and `ci.yml` to have better names?

File: .github/workflows/e2e-workflow.yaml (hunk 3 above)
Comment (imheresamir) [reply .github/workflows/e2e-workflow.yaml:10]: `master` branch will soon be renamed to `main` so I think this is fine as is

Comment (orenbm) [reply .github/workflows/e2e-workflow.yaml:10]: then the same question with `main` :)

in `main.yml` we have this guard so PRs to other branches do not run. i think that we should be aligned between GH actions, unless we have a good reason not to be. I'm ok with not having this guard but let's be consistent across actions so let's remove it from there.

**Comment by orenbm** [comment]
@NaomiHaser thanks for making the alignments!

**Review by imheresamir — APPROVED** [review]
LGTM!

**Pull request merged**